src/api/search.ts

```diff
@@ -116,40 +124,91 @@ const getProductSearch = async ({
     sort
   );
 
-  window.adobeDataLayer.push((dl : any) => {
-    dl.push({ event: 'search-request-sent', eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID } })
-  })
+  window.adobeDataLayer.push((dl: any) => {
+    dl.push({
+      event: 'search-request-sent',
+      eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID },
+    });
+  });
   // ======  end of data collection =====
 
-  const query = categorySearch && categoryId ? CATEGORY_QUERY : PRODUCT_SEARCH_QUERY;
+  const query =
+    categorySearch && categoryId ? CATEGORY_QUERY : PRODUCT_SEARCH_QUERY;
+
   const results = await fetch(apiUrl, {
     method: 'POST',
     headers,
     body: JSON.stringify({
-      query: query.replace(/(?:\r\n|\r|\n|\t|[\s]{4})/g, ' ').replace(/\s\s+/g, ' '),
+      query: query
+        .replace(/(?:\r\n|\r|\n|\t|[\s]{4})/g, ' ')
+        .replace(/\s\s+/g, ' '),
       variables: { ...variables },
     }),
   }).then((res) => res.json());
 
+  const items = results?.data?.productSearch?.items ?? [];
+
+  const productIds = items.map((item: Product) => item.product.id);
+
+  const productLabelsResults = await getGraphQL(GET_PRODUCT_LABELS_QUERY, {
+    productIds,
+    mode: 'CATEGORY',
+  });
+
+  const labels = productLabelsResults?.data?.wilsonAmLabelProvider.items ?? [];
+
+  const labelLookup: LabelLookup = labels.reduce(
+    (acc: LabelLookup, label: Label) => {
+      if (!acc[label.product_id]) {
+        acc[label.product_id] = [];
+      }
+      acc[label.product_id].push(label);
+      return acc;
+    },
+    {}
+  );
+
+  const itemsWithLabels = items.map((item: Product) => {
+    const productId = item.product.id;
+    return {
+      ...item,
+      labels: labelLookup[productId] || [],
+    };
+  });
+
+  // Replace the data in results.data.productSearch.items with itemsWithLabels
+  if (results?.data?.productSearch) {
+    results.data.productSearch.items = itemsWithLabels;
+  }
+
   // ======  initialize data collection =====
   updateSearchResultsCtx(
     SEARCH_UNIT_ID,
     searchRequestId,
     results?.data?.productSearch
   );
 
-  window.adobeDataLayer.push((dl : any) => {
-    dl.push({ event: 'search-response-received', eventInfo: { ...dl.getState(), searchUnitId: SEARCH_UNIT_ID } })
+  window.adobeDataLayer.push((dl: any) => {
```

Comment:
I think this should be done only for search query instead of both queries combined
@chicharr WDYT ?

Reply:
Here we are just triggering an event to track that the search engine responded. Tracking it before or after the extra query to get the labels would not make much difference. This is not used to track the performance/timing, just to track the actions